spring-cloud-stream-binder-kafka/src/test/java/org/springframework/cloud/stream/binder/kafka/KafkaBinderTests.java

```diff
@@ -1187,6 +1198,46 @@ public void testPartitionCountNotReduced() throws Exception {
 		assertThat(partitionSize(testTopicName)).isEqualTo(6);
 	}
 
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testConsumerDefaultDeserializer() throws Exception {
+		KafkaBinderConfigurationProperties configurationProperties = createConfigurationProperties();
+		DirectFieldAccessor consumerAccessor = new DirectFieldAccessor(getKafkaConsumer(configurationProperties));
+		assertTrue(consumerAccessor.getPropertyValue("keyDeserializer") instanceof  ByteArrayDeserializer);
+		assertTrue(consumerAccessor.getPropertyValue("valueDeserializer") instanceof  ByteArrayDeserializer);
+	}
+
+	@Test
+	@SuppressWarnings("unchecked")
+	public void testConsumerCustomDeserializer() throws Exception {
```

Comment:
It is not enough to test that overriding the properties works. We need to test that sending and receiving messages with these settings actually works.